### PR TITLE
feat: add size

### PR DESCRIPTION
**Pull request by WJswag**
memory db doesn`t expose a function that we could evaluate its size, so we change to 

- calculate converted log string lentgh if not available 
- add new log length per insert
- reset if delete all
- ignore update/delete per log change

### CODE REVIEW

File: lib/data/local/dao.dart [hunk 1]
```diff
@@ -11,13 +11,41 @@ abstract class Dao {
...
       intMapStoreFactory.store(DBConstants.FCURSOR_STORE_NAME);
+  int _JsonStringLength = -1;
+
+  int get jsonStringLength => _JsonStringLength;
 
   // Private getter to shorten the amount of code needed to get the
   // singleton instance of an opened database.
   Future<Database> get db;
 
+  Future<int> _initDBStringSizeIfRequired({bool force = false}) async {
```
Comment (jayswagchen) [inline lib/data/local/dao.dart]: `return _JsonStringLength` 沒用到的話，不如直接把這method改成Future<void>?

File: lib/data/local/dao.dart [hunk 2]
```diff
@@ -11,13 +11,41 @@ abstract class Dao {
...
+    if (force || _JsonStringLength <= 0) {
+      int s = 0;
+      return getAllLogs().then((logs) {
+        logs.forEach((element) {
+          s += element.toJson().toString().length;
+        });
+        _JsonStringLength = s;
+        return _JsonStringLength;
+      });
```
Comment (jayswagchen) [inline lib/data/local/dao.dart]: getAllLogs().then((logs)=>_calculateLog(logs))?

**Review by PeishanYen — APPROVED** [review]

**Review by jayswagchen — APPROVED** [review]

File: lib/data/local/dao.dart (hunk 1 above)
Comment (WJswag) [reply lib/data/local/dao.dart]: done

File: lib/data/local/dao.dart (hunk 2 above)
Comment (WJswag) [reply lib/data/local/dao.dart]: done

**Pull request merged**